src/sql/src/plan/statement/ddl.rs

```diff
@@ -1118,11 +1118,17 @@ fn get_encoding(
 }
 
 fn host_config(
+    scx: &StatementContext,
     remote: Option<String>,
     size: Option<String>,
 ) -> Result<StorageHostConfig, PlanError> {
     match (remote, size) {
-        (None, None) => Ok(StorageHostConfig::Undefined),
+        (None, None) => {
+            if !scx.unsafe_mode() {
+                sql_bail!("SIZE option is required");
```

Comment:
Should we make this more user-friendly with a hint about what size to pick, like @petrosagg suggested [here](https://materializeinc.slack.com/archives/CU7ELJ6E9/p1665388342309689)?